src/peft/tuners/tuners_utils.py

```diff
@@ -160,6 +161,21 @@ def active_adapters(self) -> list[str]:
     def forward(self, *args: Any, **kwargs: Any):
         return self.model.forward(*args, **kwargs)
 
+    def _pre_injection_hook(self, model: nn.Module, config: PeftConfig, adapter_name: str) -> None:
+        r"""
+        A hook to be called before the adapter is injected into the model. This method can be overridden by child
+        classes to perform any pre-injection operations.
+
+        Args:
+            model (`nn.Module`):
+                The model to be adapted.
+            config (`PeftConfig`):
+                The adapter config.
+            adapter_name (`str`):
+                The adapter name.
+        """
+        pass
```

Comment:
[nit] why not `raise NotImplementedError`? Avoid silent failures if something incorrectly calls the hook.